solr/core/src/java/org/apache/solr/handler/component/RealTimeGetComponent.java

```diff
@@ -885,7 +884,7 @@ private static SolrInputDocument toSolrInputDocument(SolrDocument doc, IndexSche
           if (!fieldArrayListCreated && doc.getFieldValue(fname) instanceof Collection) {
             // previous value was array so we must return as an array even if was a single value
             // array
-            out.setField(fname, Lists.newArrayList(val));
+            out.setField(fname, new ArrayList<>(List.of(val)));
```

Comment:
Can't just create new ArrayList with a single value. Interestingly this setField requires the list to be mutable - this might be a bug by itself. I wouldn't expect this parameter needs to be mutable. I'll see if I can clean it up. I'd prefer just `List.of()` with no ArrayList.

Reply:
Ok I see.
This method looks more complicated than it needs to be.  I don't think this condition needs to put "val" into the "out" doc, it could probably put a new ArrayList and then at the end of the loop over values, addField will be called, and this doc will be added to the existing value.  If I'm too confusing, I could try contributing this change.

Reply:
when I remove the `new ArrayList<>()` This is the error

```
  2> 162085 ERROR (qtp339291692-2438) [n:127.0.0.1:51699_solr c:org.apache.solr.cloud.NestedShardedAtomicUpdateTest_collection s:shard1 r:core_node6 x:org.apache.solr.cloud.NestedShardedAtomicUpdateTest_collection_shard1_replica_n2] o.a.s.h.RequestHandlerBase java.lang.UnsupportedOperationException
  2>           => java.lang.UnsupportedOperationException
  2>    at java.base/java.util.ImmutableCollections.uoe(ImmutableCollections.java:142)
  2> java.lang.UnsupportedOperationException: null
  2>    at java.util.ImmutableCollections.uoe(ImmutableCollections.java:142) ~[?:?]
  2>    at java.util.ImmutableCollections$AbstractImmutableCollection.add(ImmutableCollections.java:147) ~[?:?]
  2>    at org.apache.solr.common.SolrInputField.addValue(SolrInputField.java:95) ~[solr-solrj-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT 735623be5eea82a70693aee39819f2baf151aa75 [snapshot build, details omitted]]
  2>    at org.apache.solr.common.SolrInputDocument.addField(SolrInputDocument.java:104) ~[solr-solrj-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT 735623be5eea82a70693aee39819f2baf151aa75 [snapshot build, details omitted]]
  2>    at org.apache.solr.handler.component.RealTimeGetComponent.toSolrInputDocument(RealTimeGetComponent.java:892) ~[main/:?]
  2>    at org.apache.solr.handler.component.RealTimeGetComponent.toSolrInputDocument(RealTimeGetComponent.java:883) ~[main/:?]
  2>    at org.apache.solr.handler.component.RealTimeGetComponent.getInputDocument(RealTimeGetComponent.java:794) ~[main/:?]
  2>    at org.apache.solr.update.processor.DistributedUpdateProcessor.getUpdatedDocument(DistributedUpdateProcessor.java:789) ~[main/:?]
  2>    at org.apache.solr.update.processor.DistributedUpdateProcessor.doVersionAdd(DistributedUpdateProcessor.java:407) ~[main/:?]
  2>    at org.apache.solr.update.processor.DistributedUpdateProcessor.lambda$versionAdd$0(DistributedUpdateProcessor.java:357) ~[main/:?]
  2>    at org.apache.solr.update.VersionBucket.runWithLock(VersionBucket.java:51) ~[main/:?]
  2>    at org.apache.solr.update.processor.DistributedUpdateProcessor.versionAdd(DistributedUpdateProcessor.java:354) ~[main/:?]
  2>    at org.apache.solr.update.processor.DistributedUpdateProcessor.processAdd(DistributedUpdateProcessor.java:236) ~[main/:?]
  2>    at org.apache.solr.update.processor.DistributedZkUpdateProcessor.processAdd(DistributedZkUpdateProcessor.java:279) ~[main/:?]
  2>    at org.apache.solr.update.processor.LogUpdateProcessorFactory$LogUpdateProcessor.processAdd(LogUpdateProcessorFactory.java:111) ~[main/:?]
  2>    at org.apache.solr.handler.loader.JavabinLoader$1.update(JavabinLoader.java:123) ~[main/:?]
  2>    at org.apache.solr.client.solrj.request.JavaBinUpdateRequestCodec$StreamingCodec.readOuterMostDocIterator(JavaBinUpdateRequestCodec.java:342) ~[solr-solrj-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT 735623be5eea82a70693aee39819f2baf151aa75 [snapshot build, details omitted]]
  2>    at org.apache.solr.client.solrj.request.JavaBinUpdateRequestCodec$StreamingCodec.readIterator(JavaBinUpdateRequestCodec.java:286) ~[solr-solrj-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT 735623be5eea82a70693aee39819f2baf151aa75 [snapshot build, details omitted]]
  2>    at org.apache.solr.common.util.JavaBinCodec.readObject(JavaBinCodec.java:338) ~[solr-solrj-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT 735623be5eea82a70693aee39819f2baf151aa75 [snapshot build, details omitted]]
  2>    at org.apache.solr.common.util.JavaBinCodec.readVal(JavaBinCodec.java:283) ~[solr-solrj-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT 735623be5eea82a70693aee39819f2baf151aa75 [snapshot build, details omitted]]
  2>    at org.apache.solr.client.solrj.request.JavaBinUpdateRequestCodec$StreamingCodec.readNamedList(JavaBinUpdateRequestCodec.java:236) ~[solr-solrj-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT 735623be5eea82a70693aee39819f2baf151aa75 [snapshot build, details omitted]]
  2>    at org.apache.solr.common.util.JavaBinCodec.readObject(JavaBinCodec.java:303) ~[solr-solrj-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT 735623be5eea82a70693aee39819f2baf151aa75 [snapshot build, details omitted]]
  2>    at org.apache.solr.common.util.JavaBinCodec.readVal(JavaBinCodec.java:283) ~[solr-solrj-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT 735623be5eea82a70693aee39819f2baf151aa75 [snapshot build, details omitted]]
  2>    at org.apache.solr.common.util.JavaBinCodec.unmarshal(JavaBinCodec.java:193) ~[solr-solrj-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT 735623be5eea82a70693aee39819f2baf151aa75 [snapshot build, details omitted]]
  2>    at org.apache.solr.client.solrj.request.JavaBinUpdateRequestCodec.unmarshal(JavaBinUpdateRequestCodec.java:126) ~[solr-solrj-10.0.0-SNAPSHOT.jar:10.0.0-SNAPSHOT 735623be5eea82a70693aee39819f2baf151aa75 [snapshot build, details omitted]]
  2>    at org.apache.solr.handler.loader.JavabinLoader.parseAndLoadDocs(JavabinLoader.java:135) ~[main/:?]
  2>    at org.apache.solr.handler.loader.JavabinLoader.load(JavabinLoader.java:74) ~[main/:?]
  2>    at org.apache.solr.handler.UpdateRequestHandler$1.load(UpdateRequestHandler.java:101) ~[main/:?]
  2>    at org.apache.solr.handler.ContentStreamHandlerBase.handleRequestBody(ContentStreamHandlerBase.java:84) ~[main/:?]
  2>    at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:224) [main/:?]
  ...
```

Reply:
Your PR is faithful to the PR's purpose and it works; lets merge it.  Fiddling with this method too much should be separate if we want to bother.